### PR TITLE
Quick order rewamp with some tweaks

### PR DESCRIPTION
**Pull request by Danilcus-W**

### CODE REVIEW

File: code/modules/mob/living/carbon/human/ai/action_datums/fire_at_target.dm [hunk 1]
```diff
@@ -101,7 +101,8 @@
 		if(tile.density)
 			return FALSE
 
-		if(locate(/obj/structure/machinery/door) in tile)
+		var/obj/door = locate(/obj/structure/machinery/door) in tile
+		if(door && door.density)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/ai/action_datums/fire_at_target.dm]:
```suggestion
		if(door?.density)
```

File: code/modules/mob/living/carbon/human/ai/brain/ai_brain_cover.dm [hunk 2]
```diff
@@ -19,6 +19,9 @@
 	if(!COOLDOWN_FINISHED(src, cover_search_cooldown))
 		return
 
+	if(!cover_without_gun && !primary_weapon)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/ai/brain/ai_brain_cover.dm]: for some reason this feels off to me but i can't put my finger on it, so here's what all the logic does, make sure it's what you intended
```
!TRUE AND !TRUE - We can cover without guns and we have a gun, allows us to find cover
!TRUE AND !FALSE - We can cover without guns and we don't have a gun, allows us to find cover
!FALSE AND !TRUE - We can't cover without guns and we have a gun, allows us to find cover
!FALSE AND !FALSE - We can't cover without guns and we don't have a gun, stops us from finding cover
```

File: code/modules/mob/living/carbon/human/ai/brain/ai_brain_helpers.dm [hunk 3]
```diff
@@ -141,11 +141,12 @@
 
 /datum/human_ai_brain/proc/unholster_any_weapon()
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/ai/brain/ai_brain_helpers.dm:142]: switching to help intent means they're gonna get stuck by shuffling into each other a lot

Comment (Danilcus-W) [reply code/modules/mob/living/carbon/human/ai/brain/ai_brain_helpers.dm:142]: True, but also they are trying to execute things when in disarm or PB when in harm. At this moment we might unironically switch to grab intent for melee combat

File: code/modules/mob/living/carbon/human/ai/brain/ai_brain_cover.dm (hunk 2 above)
Comment (Danilcus-W) [reply code/modules/mob/living/carbon/human/ai/brain/ai_brain_cover.dm]: Somehow this seemed alright ingame, but now i can say that the logic is indeed faulty

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**